src/torrent.cpp

```diff
@@ -302,7 +302,10 @@ namespace libtorrent
 			e.fail_limit = 0;
 			e.source = announce_entry::source_magnet_link;
 			e.tier = tier;
-			m_trackers.push_back(e);
+			if (!find_tracker(e.url))
```

Comment:
wouldn't it make more sense to call ``add_tracker()`` here, instead of adding it directly to the vector?